### PR TITLE
feat: Export RegisterObjectKind and MatcherFunc from objectkinds package

### PR DESCRIPTION
**Pull request by heckler1**
This allows consumers to register arbitrary matchers for arbitrary objectkinds. This enables expanding beyond the builtins, as well as linting custom resources.

### CODE REVIEW

**Comment by janisz** [comment]
@heckler1 It will be great if you add some docs about how to use it to extend kube-linter beyond builtins

**Comment by heckler1** [comment]
Hey @janisz, good idea! I added a section to the Configuring Kube-Linter document, it seemed like a good place. I also went back and added doc comments to the exported types. Let me know what you think, open to any feedback.

File: docs/configuring-kubelinter.md [hunk 1]
```diff
@@ -142,3 +148,64 @@ For example,
...
+    // This function can contain any arbitrary logic to match objects we want to check with this lint
+    objectkinds.RegisterObjectKind(Certificate, objectkinds.MatcherFunc(func(gvk schema.GroupVersionKind) bool {
+      return gvk == certificateGVK
+    }))
+  }
+  ```
+
+- Finally, the check template itself
+
+  ```go
```
Comment (janisz) [inline docs/configuring-kubelinter.md:187]: I wonder if we could add this as [testable example](https://go.dev/blog/examples) to get fast feedback if we broke something and need to update docs.

**Comment by janisz** [comment]
Nice!

**Review by janisz — APPROVED** [review]

File: docs/configuring-kubelinter.md (hunk 1 above)
Comment (heckler1) [reply docs/configuring-kubelinter.md]: Cool, T had no idea that was a feature. I've rewritten the doc into a go file with the original code plus a quick test, and changed the markdown to refer to the go file. The example will be tested on `make test`, so we can be notified of any regressions.

**Review by janisz — APPROVED** [review]

**Comment by janisz** [comment]
Cool, Thanks!

**Pull request merged**